### ISSUE TITLE
[Grid - Drag and Drop] Interaction column has empty table header

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2879]
### Provide a general summary of the issue here

Drag and drop variants have their drag and drop "button" in their first column, but the column is missing a descriptive header. This makes the function, and the structure of the table hard to interpret without exploration.

<img width="1073" height="395" alt="Table header cell, which is the first element in the table for a drag and droppable variant of grid, is empty." src="https://github.com/user-attachments/assets/d3987fc2-ba9d-4a0c-b5ff-22bf3df60902" />

### 🤔 Expected Behavior?

Table header has a text, which describes the purpose of the column. 

### 😯 Current Behavior

Table header is empty and is not communicated to assistive technologies.

### 💁 Possible Solution

Add a text (e.g., only visible to screen reader users) which describes the purpose of the column. E.g., add a sr-only text "Reorder" to explain the purpose of the column.

### 🖥️ Steps to Reproduce

Navigate to the first cell using screen reader -> Screen reader either reads "empty" or "space" (along with mentioning the table rows/columns and announcing that the cell is a table header).   

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by DanielleRameau** [comment]
text should be hidden in the UI but should work on screenreaders. (use text reorder)
